File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool check(string& newstr, int l, int r) {

        if (l > r) {
            return true;
        }

        if (newstr[l] != newstr[r]) {
            return false;
        }
        return check(newstr, l + 1, r - 1);
    }

    bool isPalindrome(string s) {
        string newstr = "";

        for (char i : s) {
            if (isalnum(i)) {
                newstr += tolower(i);
            }
        }
        return check(newstr, 0, newstr.length() - 1);
    }
};

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int i = 0;
        int j = s.length() - 1;

        String c = s.toLowerCase();

        while (i < j) {
            while (i < j && !Character.isLetterOrDigit(c.charAt(i))) {
                i++;
            }

            while (i < j && !Character.isLetterOrDigit(c.charAt(j))) {
                j--;
            }

            if (c.charAt(i) != c.charAt(j)) {
                return false;
            }
            j--;
            i++;
        }
        return true;
    }
}